tiles/src/main/java/com/protomaps/basemap/layers/Pois.java

```diff
@@ -3,85 +3,308 @@
+          // These are duplicate of what's in the kind_detail tag
           .setAttr("cuisine", sf.getString("cuisine"))
           .setAttr("religion", sf.getString("religion"))
-          .setAttr("tourism", sf.getString("tourism"))
+          .setAttr("sport", sf.getString("sport"))
+          // Special airport only tag (to indicate if it's an airport with regular commercial flights)
           .setAttr("iata", sf.getString("iata"))
-          .setZoomRange(13, 15)
+          .setZoomRange(min_zoom, 15)
           .setBufferPixels(128);
 
-        OsmNames.setOsmNames(poly_label_position, sf, 0);
+        // Core Tilezen schema properties
+        if( kind_detail != "") {
+          poly_label_position.setAttr("pmap:kind_detail", kind_detail);
+        }
 
-        poly_label_position.setAttr("pmap:kind", kind);
-      } else if (sf.isPoint()) {
+        OsmNames.setOsmNames(poly_label_position, sf, 0);
+      } else if( sf.isPoint() ){
         var point_feature = features.point(this.name())
+          // all POIs should receive their IDs at all zooms
+          // (there is no merging of POIs like with lines and polygons in other layers)
           .setId(FeatureId.create(sf))
+          // Core Tilezen schema properties
+          .setAttr("pmap:kind", kind)
+          // While other layers don't need min_zoom, POIs do for more predictable client-side label collisions
+          // 512 px zooms versus 256 px logical zooms
+          .setAttr("pmap:min_zoom", min_zoom+1)
+          // Core OSM tags for different kinds of places
           .setAttr("amenity", sf.getString("amenity"))
-          .setAttr("shop", sf.getString("shop"))
+          .setAttr("craft", sf.getString("craft"))
+          .setAttr("historic", sf.getString("historic"))
+          .setAttr("leisure", sf.getString("leisure"))
           .setAttr("railway", sf.getString("railway"))
+          .setAttr("shop", sf.getString("shop"))
+          .setAttr("tourism", sf.getString("tourism"))
+          // Extra OSM tags for certain kinds of places
+          // These are duplicate of what's in the kind_detail tag
           .setAttr("cuisine", sf.getString("cuisine"))
           .setAttr("religion", sf.getString("religion"))
-          .setAttr("tourism", sf.getString("tourism"))
+          .setAttr("sport", sf.getString("sport"))
+          // Special airport only tag (to indicate if it's an airport with regular commercial flights)
           .setAttr("iata", sf.getString("iata"))
-          .setZoomRange(13, 15)
+          .setZoomRange(min_zoom, 15)
           .setBufferPixels(128);
 
+        // Core Tilezen schema properties
+        if( kind_detail != "") {
+          point_feature.setAttr("pmap:kind_detail", kind_detail);
+        }
+
         OsmNames.setOsmNames(point_feature, sf, 0);
 
-        point_feature.setAttr("pmap:kind", kind);
+        // Some features should only be visible at very late zooms when they don't have a name
```

Comment:
There were still too many features at zoom 15, so annotate some for later zooms. 

Should be combined with a style layer filter and style collision property.